File: JLC/d6/main.py
def get_lines_from_file(filename):
    with open(filename) as f:
        return [line.rstrip() for line in f] + [""]


def parse_lines_d2(lines):
    count = 0
    group_set = set()
    new_person = True
    for line in lines:
        if line == "":
            print(len(group_set))
            count += len(group_set)
            group_set.clear()
            print("----")
            new_person = True
            continue
        else:
            if new_person:
                group_set = set(line)
            else:
                group_set &= set(line)  # This is the reason I went Python :)
            print(group_set)
            new_person = False
    print(count)


def main():
    lines = get_lines_from_file("foo.txt")
    parse_lines_d2(lines)


if __name__ == "__main__":
    main()

File: JLC/d6/main.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <unordered_set>
#include <vector>
using namespace std;


vector<string>* get_lines_from_file(string filename) {
    ifstream myfile(filename);
    string line;
    vector<string>* v = new vector<string>;
    while (getline(myfile, line)) {
        v->push_back(line);
    }
    return v;
}


void parse_lines_p1(vector<string>* lines) {
    int count = 0;
    unordered_set<char> set = unordered_set<char>();
    for (string line : *lines) {
        if (line == ""){
            cout << set.size() << endl;
            count += set.size();
            set.clear();
            continue;
        }
        for (char c : line) {
            set.insert(c);
        }
    }
    cout << count << endl;
}


int main() {
    vector<string>* lines = get_lines_from_file("foo.txt");
    lines->push_back("");
    parse_lines_p1(lines);
}
